### PR TITLE
Add support for x-apigw-api-id request header in Glue

### PR DESCRIPTION
**Pull request by nevillelyh**
## Description

Add support for x-apigw-api-id request header in Glue

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Section
* Add support for `x-apigw-api-id` request header in AWS Glue via the `hive.metastore.glue.proxy-api-id` property
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java [hunk 1]
```diff
@@ -502,6 +503,19 @@ public HiveConfig setMaxPartitionBatchSize(int maxPartitionBatchSize)
         return this;
     }
 
+    public Optional<String> getGlueProxyApiId()
+    {
+        return glueProxyApiId;
+    }
+
+    @Config("hive.metastore.glue.proxy-api-id")
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:511]: Why not `GlueHiveMetastoreConfig`? Also, could you update "AWS Glue catalog configuration properties" in hive.rst? 

(I know it comes internal code though)

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java [hunk 2]
```diff
@@ -57,12 +63,21 @@
...
-    public void configure(Binder binder)
+    public void setup(Binder binder)
     {
-        configBinder(binder).bindConfig(HiveConfig.class);
+        HiveConfig hiveConfig = buildConfigObject(HiveConfig.class);
+        hiveConfig.getGlueProxyApiId().ifPresent(glueProxyApiId -> {
+            GlueHiveMetastoreConfig glueConfig = buildConfigObject(GlueHiveMetastoreConfig.class);
+            checkState(
+                    glueConfig.getGlueRegion().isPresent() && glueConfig.getGlueEndpointUrl().isPresent(),
+                    "Both Glue region and Glue endpoint URL must be provided when Glue proxy API ID is present");
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java:76]: We can use `@AssertTrue` or `@PostConstruct` annotations within the config class likes `MySqlJdbcConfig` & `BigQueryConfig` for verifying the config values once you moved to Glue config class.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/ProxyApiRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @Override
+    public void beforeRequest(Request<?> request)
+    {
+        // AWS Glue SDK will append "X-Amz-Target" header to requests (with "AWSGlue" prefix).
+        // This misleads API Gateway (Glue proxy) that it's not the target of the REST call. Therefore we
+        // need to pass "X-Amz-Target" value in a special HTTP header that is translated back to "X-Amz-Target"
+        // when API Gateway makes request to AWSGlue.
+        request.getHeaders().put("X-Starburst-Amz-Target-Proxy", request.getHeaders().remove("X-Amz-Target"));
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/ProxyApiRequestHandler.java:46]: This not Starburst product.

**Review by ebyhr — CHANGES_REQUESTED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java (hunk 1 above)
Comment (nevillelyh) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:511]: Totally missed that. Thanks.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/ProxyApiRequestHandler.java (hunk 3 above)
Comment (nevillelyh) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/ProxyApiRequestHandler.java:46]: Fixed.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java (hunk 2 above)
Comment (nevillelyh) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java]: Fixed.

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/connector/hive.rst [hunk 4]
```diff
@@ -654,6 +654,8 @@ Property Name                                        Description
                                                      required when running in a GovCloud region.
                                                      Example: ``us-gov-east-1``
 
+``hive.metastore.glue.proxy-api-id``                 ID of Glue Proxy API.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/hive.rst:657]: very sparse docs - I'm not sure if I'll be able to tell when to set this or not. Can you mention when one might want to set this and to what value?

Comment (nevillelyh) [reply docs/src/main/sphinx/connector/hive.rst]: Yeah very sparse even in AWS docs. Added a little but probably only make sense to those who know AWS VPC & networking.

**Review by hashhar — APPROVED** [review]
LGTM from me. Will defer to @ebyhr on the merge.

**Review by kokosing — APPROVED** [review]

**Pull request merged**